binderhub/repoproviders.py

```diff
@@ -356,3 +364,54 @@ def get_resolved_ref(self):
 
     def get_build_slug(self):
         return '{user}-{repo}'.format(user=self.user, repo=self.repo)
+
+
+class GitHubGistRepoProvider(GitHubRepoProvider):
+    """GitHub gist provider.
+
+    Users must provide a spec that matches the following form (similar to github)
+
+    <username>/<gist-id>[/<ref>]
```

Comment:
Does this mean the ref is optional? I'd ideally like it to not be optional in the URL (we can ofc construct it in the JS). Git cloning says the default branch is master, so we should require a ref and go through the usual process of resolving it in the backend IMO.

Reply:
so there is no branch called `master` as far as the upstream gist repo is concerned.

Reply:
we can use `master` as a sentinal value meaning latest in the history chain if we want

Reply:
Ah, hmm. I did a git clone and got to 'master'. Perhaps that's git clone's behavior? Am unsure. Am +1 using master as sentinel if that's not the case.